deviceModeInit.js

```diff
@@ -171,11 +176,42 @@ var rudderTracking = (function () {
     };
     if (pages[path] === "Registration Viewed") {
       rudderanalytics.track(pages[path], properties);
+      rs$("#create_customer").submit(userRegistered);
+    } else if (pages[path] === "Login Viewed") {
+      rudderanalytics.track(pages[path], properties);
+      rs$("#customer_login").submit(userLoggedIn);
     } else {
       rudderanalytics.page(category, pageName, properties);
     }
   }
 
+
+  function userRegistered() {
+    const email = rs$('#create_customer [type="email"]').val();
+    const firstName = rs$(
+      '#create_customer [name="customer[first_name]"]'
+    ).val();
+    const lastName = rs$('#create_customer [name="customer[last_name]"]').val();
+    rudderanalytics.identify({
+      email,
+      firstName,
+      lastName,
+    });
```

Comment:
Here can't we set emai as userId?

Reply:
As we are setting `customerId` as `userId` in cloud-mode for identity stitching we are maintaining same parity here

Reply:
We have `__st.cid` that is a shopify generated id for a customer, that is also used in the server-side events. Using the same id here will allow for a complete user journey tracking from both server and client side. @utsabc I believe this is the reason?